tamr_unify_client/client.py

```diff
@@ -1,62 +1,76 @@
+import logging
+from typing import Optional
 from urllib.parse import urljoin
 
 import requests
-from requests import Response
+import requests.auth
+import requests.exceptions
 
 from tamr_unify_client.dataset.collection import DatasetCollection
 from tamr_unify_client.project.collection import ProjectCollection
 
-# monkey-patch Response.successful
+logger = logging.getLogger(__name__)
 
 
-def successful(self):
-    """Checks that this response did not encounter an HTTP error (i.e. status code indicates success: 2xx, 3xx).
+def successful(response: requests.Response) -> requests.Response:
+    """Checks that this response did not encounter an HTTP error.
 
-    :raises :class:`requests.exceptions.HTTPError`: If an HTTP error is encountered.
-    :return: The calling response (i.e. ``self``).
-    :rtype: :class:`requests.Response`
+    HTTP error codes match 4xx or 5xx.
+
+    Returns:
+        The response being checked.
+
+    Raises:
+        requests.exceptions.HTTPError: If an HTTP error is encountered.
```

Comment:
per `requests.Response.raise_for_status`, this can raise HTTPError or URLError
https://2.python-requests.org//en/v0.6.5/api/#requests.Response.raise_for_status

Reply:
Those docs are out-of-date. The stable docs are here: https://requests.readthedocs.io/en/master/

In these [docs](https://requests.readthedocs.io/en/master/api/#requests.Response.raise_for_status) and in the [source code](https://requests.readthedocs.io/en/master/_modules/requests/models/#Response.raise_for_status) for `raise_for_status`, `URLError` is not raised.